sdk/core/Azure.Core/src/Pipeline/LoggingPolicy.cs

```diff
@@ -274,6 +199,215 @@ protected override void Dispose(bool disposing)
 
                 _originalStream.Dispose();
             }
+
+            private void DecrementLength(ref int count)
+            {
+                var left = Math.Min(count, _maxLoggedBytes);
+                count = left;
+                _maxLoggedBytes -= count;
```

Comment:
Nah, to make it truly thread-safe you also need to avoid data race in                  `var left = Math.Min(count, _maxLoggedBytes);` but if you are making unsynchronized reads from the stream you have bigger problems than slightly longer logs.